### ISSUE TITLE
Cloning Repo with Invalid Filename fails on Windows

### ISSUE THREAD

**Issue by jojobyte** [nodegit/nodegit#1094]
Trying to clone the repo from https://github.com/bocoup/opendesignkit on windows and I get an empty error object.

I'm fairly certain its because of the colon in this file: https://github.com/bocoup/opendesignkit/blob/master/wiki/Open-Design-Kit:-Design-Brief.md

I cloned the repo with TortoiseGit and it worked but had a changed file immediately after clone, The above file is missing and "Open-Design-Kit" with no extension is in its place.

Presumably that is a vaild filename on mac/linux, is there a workaround for something like this with nodegit as clearly TortoiseGit had one?

**Comment by tbranyen** [comment]
Looks to be an issue with libgit2, as reported here: https://github.com/libgit2/libgit2/issues/2994

**Comment by saper** [comment]
This is a well-known incompatibility with NTFS, since colon on Windows means either a drive letter separator or indicates the alternate file stream (see http://latkin.org/blog/2016/07/20/git-for-windows-accidentally-creates-ntfs-alternate-data-streams/ for more details). I have seen this problem with some SVN clients, so this problem is by no means limited to git.

**Comment by jojobyte** [comment]
Makes sense, even if there isnt a fix for nodegit, if it returned more than an empty error object it would be handy.

Tested it with `git clone https://github.com/bocoup/opendesignkit.git` just now and it performs the same way as TortoiseGit.

Shouldn't libgit2 perform the same way as the actual CLI or is this some pre-checkout windows specific hook the CLI installs that libgit2 is missing?
